packages/core/src/websockets/reconnecting-websocket.ts

```diff
@@ -121,24 +122,27 @@ export type Message = string | ArrayBuffer | Blob | ArrayBufferView;
 
 let didWarnAboutMissingWebSocket = false;
 
-export class ReconnectingWebSocket extends TypedEventTarget<WebSocketEventMap> implements IReconnectingWebSocket {
+export class ReconnectingWebSocket<BType extends string = BinaryType>
```

Comment:
I'm not very familiar with `WebSocket.BinaryType`, but the fact that it is both 1) used a type generic that gets narrowed in the constructor and 2) is mutable within this class feels a bit off which I think is evident by all the type-casting that is being done around `BType`. Would it be reasonable to get rid of the generic and just have the binaryType getter and setter functions just work with a type `string` or `BinaryType` instead?

Related, looking at the other places in our code that mention binaryType, it's unclear which BinaryType(s) should be settable. Is it really just `"arraybuffer" | "blob"` or should `"nodebuffer" | "fragments"` also be acceptable?

Reply:
It depends on the environment. In the browser `nodebuffer` and `fragments` don't exist

Reply:
The reason the typecasting is done is because the default for `BinaryType` comes from builtin typing but it assumes the browser context

Reply:
Ideally we would be able to pass the `WebSocket` class type as a generic and pull the `binaryType` from there, but the `WebSocket` global type from the built-in types does not include that information until you construct the WebSocket itself (it's just an interface)

Reply:
Okay so I was able to achieve the (almost) ideal solution by creating a generic interface with the default being the global `WebSocket` interface and reading the type of `binaryType` off of the passed in implementation. Let me know if this is better and/or if it makes more sense now what is happening

Reply:
Nice! Thank you for the explanation; looks great!